californium-core/src/main/java/org/eclipse/californium/core/CoapClient.java

```diff
@@ -855,8 +855,7 @@ private CoapObserveRelation observeAndWait(Request request, CoapHandler handler)
 		// add notification listener to all notification
 		NotificationListener notificationListener = new Adapter(messageObserver, request);
 		outEndpoint.addNotificationListener(notificationListener);
-		// messageobserver should remove this listener when the request is cancelled
-		messageObserver.setNotificationListener(notificationListener);
+		// relation should remove this listener when the request is cancelled
 		relation.setNotificationListener(notificationListener);
 		CoapResponse response = synchronous(request, outEndpoint);
```

Comment:
Up to here this is the same code as in `observe()`.
Maybe we can refactor this code into a separate private method?